internal/common/debug/example_test.go

```diff
@@ -36,8 +37,54 @@ type (
 	stopperImpl struct {
 		pollerTracker *pollerTrackerImpl
 	}
+
+	// activityTrackerImpl implements the ActivityTracker interface
+	activityTrackerImpl struct {
+		sync.RWMutex
+		m map[ActivityInfo]int64
+	}
+
+	// activityStopperImpl implements the Stopper interface
+	activityStopperImpl struct {
+		sync.Once
+		info    ActivityInfo
+		tracker *activityTrackerImpl
+	}
 )
 
+var _ ActivityTracker = &activityTrackerImpl{}
+var _ Stopper = &activityStopperImpl{}
+
+func (ati *activityTrackerImpl) Start(info ActivityInfo) Stopper {
+	ati.Lock()
+	ati.m[info]++
+	ati.Unlock()
+	return &activityStopperImpl{info: info, tracker: ati}
+}
+
+func (ati *activityTrackerImpl) Stats() Activities {
```

Comment:
We can group Activities by activityType or other fields of ActivityInfo if needed